presto/presto.go

```diff
@@ -554,17 +560,23 @@ func (st *driverStmt) QueryContext(ctx context.Context, args []driver.NamedValue
 
 	if len(args) > 0 {
 		hs = make(http.Header)
-		hs.Add(preparedStatementHeader, preparedStatementName+"="+url.QueryEscape(st.query))
-
-		ss := make([]string, len(args))
-		for i, arg := range args {
-			s, err := Serial(arg.Value)
-			if err != nil {
-				return nil, err
+		if args[0].Name == prestoUserHeader {
+			st.user = args[0].Value.(string)
```

Comment:
this will panic if the value isn't string; not sure we want this code path to panic

cc @skogtwin

Reply:
Not sure what kind of authority I am :) But yeah, this basically forces to pass "X-Presto-User" as a first argument in a slice, and will fail or even panic otherwise. It's better to search for the header through the whole `args` slice and use `res, ok` typecasting with proper error processing.

Reply:
The reason I made this change is in the preparedStatementHeader code, it only cares about the Value, I think it handles one by one to replace the '?'. Then I need to do something special for this auth thing, either to be the first or to be last. If I put random places, as a user, I will be very confusing to write something like, 

db.Query("select * from XX where a= ? and b = ?", []interface{}{1}, sql.Named("X-Presto-User", string("Alice")), []interface{}{"abc"}

Reply:
Yeah, but it would make sense to write `db.Query("select * from XX where a= ? and b = ?", []interface{}{1}, []interface{"abc"}, sql.Named("X-Presto-User", string("Alice")))`. Documenting the order could also be a solution, but I'd rather be bulletproof.

Reply:
ok, we can move this to the last

Reply:
> ok, we can move this to the last

This is not exactly what I meant :) It does not really matter is it the last or the first argument after the query itself. The point is, this position should either be documented or do not matter. If you insist on having it at a fixed potision, fine, but then QueryContext should have a doc-comment that describes it (frankly, it should have a commentary anyway, I'm surprised it does not).